packages/cli/src/commands/build/index.ts

```diff
@@ -426,6 +426,23 @@ async function doBuild(
 
   const ops: Promise<Error | void>[] = [];
 
+  const dependencies = [
+    ...Object.keys(pkg?.dependencies ?? {}),
+    ...Object.keys(pkg?.devDependencies ?? {}),
+  ];
+
+  const isUsingSpeedInsights = dependencies.some(
+    d => d === '@vercel/speed-insights'
+  );
+
+  if (isUsingSpeedInsights && process.env.VERCEL_ANALYTICS_ID) {
+    output.warn(
+      `The \`VERCEL_ANALYTICS_ID\` environment variable is deprecated and will be removed in a future release. Please remove it from your environment variables`
+    );
+
+    delete process.env.VERCEL_ANALYTICS_ID;
+  }
```

Comment:
I wrote up an example test here. It's only asserting the warning, but I think that's good enough for this logic. https://github.com/vercel/vercel/pull/10541